packages/utils/src/EmitterOf.ts

```diff
@@ -0,0 +1,35 @@
+// Utility type EmitterOf<T>. By deriving an interface from this type, 
+// you declare that the interface has the standard EventEmitter3  
+// listener setter functions 'on', 'off', 'once' for each event type
+// defined in T.
+
+// Convert EventEmitter3 event types to corresponding listener
+// setter function types (eg. types of 'on', 'off', 'once').
+// For example, 
+// message: (message: Message) => void
+// -> message: (event: 'message', listener: (message: Message) => void) => void
```

Comment:
Could use different names for function and parameter names here, so that the example is easier to understand?